### ISSUE TITLE
Adding a recording for a sound doesn't mark the project as "needs-save"

### ISSUE THREAD

**Issue by jwzimmer** [LLK/scratch-flash#1255]
Staging & Production, Chrome 55, Mac OS El Capitan

**TL;DR** Any changes are supposed to trigger an auto-save, but adding a recording does not.

To repro:
- Go to the online editor
- Click add a recording in the Sounds tab
- Note that adding the recording triggers an auto-save, but the recording part of that sound hasn't actually been created yet (duration is 0)
- Add a recording to the sound
- Wait a bit, note that "Save now" doesn't change to "saving", etc.
- Navigate away from the editor
- Note that although your changes will now being saved, you don't see the dialogue box warning you about your unsaved changes, which should happen if a project is in the needs-save state when you try to leave the editor

@cwillisf described when a project should be marked as "needs-save":
> - when you change something, the project is marked "dirty" (needs to save)
> - if you try to leave while the project is marked dirty, then pop up that warning dialog asking for confirmation
> - if at any time it's been at least 2 minutes since the last save, and the project is marked dirty, then start an autosave.
> - if the project finishes saving (either by autosave or by the user clicking on the save text) then clear the dirty flag

Given those criteria, it seems that adding the actual recording to the sound doesn't mark the project as "needs-save", but it should, since it is a change, and any change should make the project "needs-save".
